main/mainworker.cpp

```diff
@@ -11815,12 +11815,12 @@ bool MainWorker::SwitchLight(const uint64_t idx, const std::string &switchcmd, c
 		return SwitchLightInt(sd, switchcmd, level, hue, false);
 }
 
-bool MainWorker::SetSetPoint(const std::string &idx, const float TempValue, const int newMode, const std::string &until)
+bool MainWorker::SetSetPoint(const std::string &idx, const float TempValue, const std::string &newMode, const std::string &until)
 {
 	//Get Device details
 	std::vector<std::vector<std::string> > result;
 	result = m_sql.safe_query(
-		"SELECT HardwareID, DeviceID,Unit,Type,SubType,SwitchType,StrParam1 FROM DeviceStatus WHERE (ID == '%q')",
+		"SELECT HardwareID, DeviceID,Unit,Type,SubType,SwitchType,StrParam1,ID FROM DeviceStatus WHERE (ID == '%q')",
```

Comment:
I think something is duplicate here.. selecting the ID where ID ==

Reply:
That's done deliberately so that we can pass the result directly to `SetSetPointInt()` without having to go via the one-arg version of `SetSetPoint()`and do the query again.

Now if you look *really* hard you'll probably see that the only time `SetSetPointInt()` actually looks at `sd[7]` it's the code path that calls back into the three-arg Evohome `SetSetPoint()` function, so you could argue that I didn't need to add it. But then you get to go and sit on the naughty step. Ick :)

Reply:
Looking back at that, I missed the addition of HTYPE_EVOHOME_TCP when switching from master to development branches. Fixed locally; will push as soon as I've sorted out your other comment.

Reply:
Good catch! +1 for the ID